### PR TITLE
Add module centric view to Tracer viewer

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Improved UI in the viewer.
Added unit tests.

#### PR validation:

Code was tested with HLT tracer log output.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Services/web/README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ open -a Google\ Chrome --args --allow-file-access-from-files file://<where ever
...
 
 ## Viewer layout
-The viewer is composed to two main areas, the top is the timing viewer and at the bottom shows information about a selected time block. The top time viewer is further divided into three parts. On the left is the macro scoping grouping of framework activity types into Global and Stream activities. At the bottom is the measurement of time since the start of the job. The main area shows the blocks of time spend doing various work within the Framework.
+The viewer is composed to two main areas, the top is the timing viewer and at the bottom shows information about a selected time block. The top time viewer is further divided into three parts. On the left is the macro scoping grouping of framework activity types into Global and Stream activities. If the `module centric` option was chosen then this area is also divided by each module activity which is sorted based on most time used to least time used. At the bottom is the measurement of time since the start of the job. The main area shows the blocks of time spend doing various work within the Framework.
```
Comment (makortel) [inline FWCore/Services/web/README.md:25]: Would `"module centric"` or `_module centric_` provide better formatting? (I was just surprised to see the code formatting with backquotes for it)

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
The usual
* https://github.com/cms-sw/cmssw/issues/43293
* https://github.com/cms-sw/cmssw/issues/39803
* https://github.com/cms-sw/cmssw/issues/43456

**Comment by Dr15Jones** [comment]
@cms-sw/orp-l2 I have another change dependent upon this one I've just finished. It would be helpful to merge this so I can open a PR with the next change.

**Comment by Dr15Jones** [comment]
@cms-sw/orp-l2 please merge as this is holding up other developments.

**Comment by rappoccio** [comment]
+1

**Pull request merged**